### PR TITLE
Adds a texture mapping allowing us to combine sampler uniforms

### PR DESCRIPTION
**Pull request by GSterbrant**
### Description
All frontend chunks use the new textureMapping lookup to understand if the chunk needs to output a new uniform sampler for its texture fetches or if the same texture is already declared as a sampler elsewhere. This solves a bug where some materials just fails to compile due to overuse of texture samplers.

For example, the glTF sample [Stained Glass Lamp](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/StainedGlassLamp) would fail to compile with clustered on using over 16 samplers, with this PR only uses 10.

### CODE REVIEW

**Review by willeastcott — APPROVED** [review]
A great improvement that also actually removes code from the engine. Nice! Get at least one more review though before merging.

File: src/graphics/program-lib/programs/standard.js
Comment (slimbuck) [inline src/graphics/program-lib/programs/standard.js:352]: diffuseDetail doesn't necessarily have the same encoding as diffuse

**Review by slimbuck — COMMENTED** [review]
Awesome PR @GSterbrant. I especially like that existing chunks are still supported (..right?).

Few things:
- please update chunk validation with updated API for frontend chunks
- please update chunk docs on developer site

File: src/scene/materials/standard-material-options-builder.js
Comment (slimbuck) [inline src/scene/materials/standard-material-options-builder.js:159]: also need to add sheenEncoding?

File: src/graphics/program-lib/programs/standard.js
Comment (GSterbrant) [reply src/graphics/program-lib/programs/standard.js:352]: Ah! Good catch, will fix that!

Comment (GSterbrant) [reply src/graphics/program-lib/programs/standard.js]: https://github.com/playcanvas/developer.playcanvas.com/pull/439

Here's the PR for the chunk migrations.

**Review by slimbuck — APPROVED** [review]
Fricken awesome!

**Review by mvaligursky — APPROVED** [review]

**Pull request merged**
